### DISCUSSION TITLE
Twin Systems

### DISCUSSION THREAD

**Discussion by ralfHielscher** [mtex-toolbox/mtex#870] [Feature Requests]
Implement a class `twinSystem` similar to `slipSystem`. 

Usage:
* Schmid-factor
* taylor model
* chracterisation of twin boundaries
* twin reconstruction

Representation:
What is the smallest subset of properties we have to store? Needs to be evaluated!
* misrotation
* twinning plane
* twinning direction ??
* CRSS ??

**Comment by Tijmenvermeij** [comment]
Hi,

I have been playing a bit with some EBSD/TKD data that has deformation twins in alpha-Ti and it is becoming a bit more clear to me which things may be needed to handle (deformation) twinning in an easier way in Mtex.
I think this feature development will go hand in hand with the Feature Request on "twinning reconstruction": https://github.com/mtex-toolbox/mtex/discussions/871

Some thoughts:

1. Currently in the Mtex documentation, https://mtex-toolbox.github.io/TwinningBoundaries.html, an orientation relationship (OR) is found for twin boundaries by calculating the average misorientation from boundaries in the map that seem to be twin boundaries (in that case if misorientation angle is around 86degree). From that rounded Miller indices are found and an OR is generated which can then be checked for at each GB segment to ID twin boundaries. 
A strange thing to me is that the resulting Miller indices, 
```
     plane parallel   direction parallel      fit
(1-101) || (10-1-1)   [01-11] || [1-101]      0.466°
```
are not matching the description of the twinning system that is supposed to be active, the {10-12} system, see the top row of the table below.
![image](https://github.com/mtex-toolbox/mtex/assets/16989758/8b0a01a9-b293-4bc1-882e-ba3fa6888ff7)
In my view, the twinning OR should have the twinning plane and the rotation axis in the OR. But maybe I'm wrong on this. At least in when calculation the rotation axis of the OR it results in <1-210>, so that seems ok (also the angle of 86deg matches).

2. To allow the same kind of analysis as with a slip system, the twinning system would need to be symmetrized. This is of course already possible if you just treat the twinning system as a slip system in Mtex and symmetrise it, but, for each individual twinning variant/system, it will also need to be linked to the misrotation of that twinning variant. Then we can, e.g., check which twinning variant is present at a boundary segment and compare the boundary direction to the twinning plane trace.
To do this, I think we need to be able to generate an OR with `orientation.map` for a known twinning system and symmetrise it in some way while keeping track of the specific twinning plane and direction that are corresponding to each one...

3. A deformation twinning system has a specific amount of shear, which is dependent on the twinning system and the c/a ration, where does that go and is it need? I guess for a taylor model it's useful and also for identification with algorithms like SSLIP.

4. We have to be careful that most twinning systems in HCP are "unidirectional", i.e. compression of tension of c-axis, as opposed to slip which can happen in opposite directions. The direction depends on the twinning system and also sometimes again on the c/a ratio. I'm not entirely sure, but I think this also means that the rotation only occurs in one direction...
This also makes the HCP twins different from FCC/BCC twins.

5. Is it possible to use (parts of) the parentGrainReconstructor for twinning analysis? I recall someone said at the last Mtex workshop that they tried this, but not 100% sure. This could be especially useful to get the different variants.

6. Martensite transformation (TRIP, but also just thermal) also has a relatively well-defined shear, similar to twinning, @frankNiessen (I think) was also talking about this during the Mtex workshop. So maybe there needs to be a more generalized class of some sort that can cover both twinning and martensite...?

This may be a good point to start from :)

Cheers,
Tijmen
